server/src/internalClusterTest/java/org/opensearch/search/backpressure/SearchBackpressureIT.java

```diff
@@ -0,0 +1,284 @@
+                latch.countDown();
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                logger.info("testSearchShardTaskCancellationWithHighCpu onFailure e = " + e.getMessage());
+                assertEquals(TaskCancelledException.class, e.getClass());
+                assertTrue(e.getMessage().contains("cpu usage exceeded"));
+                latch.countDown();
+            }
+        });
+
+        latch.await();
+        resetSettingValue(CpuUsageTracker.SETTING_CPU_TIME_MILLIS_THRESHOLD.getKey());
+    }
+
+    public void testSearchCancellationWithBackpressureDisabled() throws InterruptedException {
+        ClusterUpdateSettingsRequest updateSettingsRequest = new ClusterUpdateSettingsRequest();
+        updateSettingsRequest.persistentSettings(Settings.builder().put(SearchBackpressureSettings.SETTING_MODE.getKey(), "monitor_only"));
+        assertAcked(client().admin().cluster().updateSettings(updateSettingsRequest).actionGet());
+
+        CountDownLatch latch = new CountDownLatch(1);
+
+        client().execute(TestTransportAction.ACTION, generateRequestWithHigCpuUsage(0.8, 10000), new ActionListener<>() {
+            @Override
+            public void onResponse(TestResponse testResponse) {
+                latch.countDown();
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                fail("SearchShardTask shouldn't have cancelled for monitor_only mode");
+                latch.countDown();
+            }
+        });
+
+        latch.await();
+        resetSettingValue(SearchBackpressureSettings.SETTING_MODE.getKey());
+    }
+
+    /**
+     * Returns a {@link TestRequest} which will continue running for given milliseconds
+     * @param duration time in milliseconds
+     */
+    private TestRequest generateRequestWithHighElapsedTime(long duration) {
+        return new TestRequest(() -> {
+            try {
+                Thread.sleep(duration);
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+            }
+        });
+    }
+
+    /**
+     * Returns a {@link TestRequest} which is CPU heavy
+     * @param load percentage load to the CPU
+     * @param duration time in milliseconds
+     */
+    private TestRequest generateRequestWithHigCpuUsage(double load, long duration) {
+        return new TestRequest(() -> {
+            long startTime = System.currentTimeMillis();
+            try {
+                while (System.currentTimeMillis() - startTime < duration) {
+                    // sleeping for the percentage of unladen time every 100ms
+                    if (System.currentTimeMillis() % 100 == 0) {
+                        Thread.sleep((long) Math.floor((1 - load) * 100));
+                    }
+                }
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+            }
+        });
+    }
```

Comment:
The same comment applies here as well. We don't really have to simulate an `x%` CPU load for `y` seconds.

We just need to execute simple instructions that consume some CPU cycles, and be small enough that it can be interleaved with the cancellation checks. The busy-wait loop already takes care of re-running this 'work' until enough CPU cycles have been consumed for this task cancellation to take place.

Reply:
Noted.